### ISSUE TITLE
Expose Jaeger agent port as environment variable on deployment manifest

### ISSUE THREAD

**Issue by masroorhasan** [SeldonIO/seldon-core#396]
The default configuration for Jaeger tracing allows specifying jaeger agent host address but hardcodes port `5775` (for zipkin.thrift over thrift protocol). 
```python
jaeger_serv = os.environ.get("JAEGER_AGENT_HOST","0.0.0.0")
        jaeger_config = os.environ.get("JAEGER_CONFIG_PATH",None)
        if jaeger_config is None:
            logger.info("Using default tracing config")
            config = Config(
                config={ # usually read from some yaml config
                    'sampler': {
                        'type': 'const',
                        'param': 1,
                    },
                    'local_agent': {
                        'reporting_host': jaeger_serv,
                        'reporting_port': 5775,
                    },
                    'logging': True,
                },
                service_name=args.interface_name,
                validate=True,
            )
```

However, Jaeger agent exposes multiple UDP ports to support jaeger.thrift and binary thrift as well: https://www.jaegertracing.io/docs/1.8/deployment/#agent

The default seldon python deployment should expose enabling the agent port through environment variable so users can pick the right port to configure clients (for example, `6831`).

Setting that with configuration YAML mounted on config map does not cover all use cases of setting host and port. For example, Jaeger agent works best deployed as a `Daemonset` on each Node, with deployments scheduled on Node only needs to send data to its local agent. Setting `status.hostIP` on the data in a config map is not supported native (could mount and `sed` update the value but that seems like a hack).

Basically, proposing to support this on a seldon deployment:
```yaml
         - name: JAEGER_AGENT_HOST
            valueFrom:
              fieldRef:
                fieldPath: status.hostIP
          - name: JAEGER_AGENT_PORT
             value: "6831"
```

Proposed change: https://github.com/SeldonIO/seldon-core/compare/master...masroorhasan:masroor/jaeger-port

If the above change is ok, I'd like to submit a PR (pending discussion on the change). I assume docs and example would need to be updated as well.

Thanks!

**Comment by cliveseldon** [comment]
Your proposed change makes sense. 
We should fully support Jaeger configuration.
Please do a PR.
